public/docs/ts/latest/tutorial/toh-pt5.jade

```diff
@@ -421,120 +503,171 @@ code-example(format='').
+  So we convert the route parameter value to a number with the !{_str2int}.
 
   ### Add *HeroService.getHero*
+
   The problem with this bit of code is that `HeroService` doesn't have a `getHero` method!
   We better fix that quickly before someone notices that we broke the app.
 
   Open `HeroService` and add a `getHero` method that filters the heroes list from `getHeroes` by `id`:
-+makeExample('toh-5/ts/app/hero.service.ts', 'get-hero', 'app/hero.service.ts (getHero)')(format=".")
+
++makeExcerpt('app/hero.service.ts', 'getHero')
+
 :marked
-  Return to the `HeroDetailComponent` to clean up loose ends.
+  Let's return to the `HeroDetailComponent` to clean up loose ends.
 
   ### Find our way back
 
   We can navigate *to* the `HeroDetailComponent` in several ways.
   How do we navigate somewhere else when we're done?
 
   The user could click one of the two links in the `AppComponent`. Or click the browser's back button.
-  We'll add a third option, a `goBack` method that navigates backward one step in the browser's history stack
+  We'll add a third option, a `goBack` method that navigates backward one step in the browser's history stack.
+
++makeExcerpt('app/hero-detail.component.ts', 'goBack')
 
-+makeExample('toh-5/ts/app/hero-detail.component.ts', 'go-back', 'app/hero-detail.component.ts (goBack)')(format=".")
+- var _CanDeactivateGuard = _docsFor == 'dart' ? '<em>routerCanDeactivate</em> hook' : '<em>CanDeactivate</em> guard'
+- var _CanDeactivateGuardUri = _docsFor == 'dart' ? 'angular2.router/CanDeactivate-class' : 'router/index/CanDeactivate-interface'
 .l-sub-section
- :marked
-   Going back too far could take us out of the application.
-   That's acceptable in a demo. We'd guard against it in a real application,
-   perhaps with the [*CanDeactivate* guard](../api/router/index/CanDeactivate-interface.html).
+  :marked
+    Going back too far could take us out of the application.
+    That's acceptable in a demo. We'd guard against it in a real application,
+    perhaps with the [!{_CanDeactivateGuard}](../api/!{_CanDeactivateGuardUri}.html).
+
 :marked
-  Then we wire this method with an event binding to a *Back* button that we add to the bottom of the component template.
-+makeExample('toh-5/ts/app/hero-detail.component.html', 'back-button')(format=".")
+  Then we wire this method with an event binding to a *Back* button that we
+  add to the bottom of the component template.
+
++makeExcerpt('app/hero-detail.component.html', 'back-button', '')
+
 :marked
-  Modifing the template to add this button spurs us to take one more incremental improvement and migrate the template to its own file
-  called `hero-detail.component.html`
-+makeExample('toh-5/ts/app/hero-detail.component.html', '', 'app/hero-detail.component.html')(format=".")
+  Modifing the template to add this button spurs us to take one more 
+  incremental improvement and migrate the template to its own file:
```

Comment:
I found it disconcerting to not have the filename mentioned here, even though it's apparently in the title of the immediately following excerpt. (Don't make me think! Draw connections explicitly.)

own file -> own file, called `hero-detail.component.html`:

(of course, you'd need to adjust the name for Dart)